app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -70,12 +81,24 @@
  * @author ctsims
  */
 @ManagedUi(R.layout.screen_login)
-public class LoginActivity extends CommCareActivity<LoginActivity> implements OnItemSelectedListener {
+public class LoginActivity extends CommCareActivity<LoginActivity>
+        implements OnItemSelectedListener, RuntimePermissionRequester {
 
     private static final String TAG = LoginActivity.class.getSimpleName();
-    
+
+    private final static String[] appPermissions =

```

Comment:
Some activities need different permissions given different states (like https://github.com/dimagi/commcare-odk/pull/818/files#diff-3e9cdafcf0cfda6187cf54c80c8115deR172). That type of permission request dispatch wouldn't be possible with annotations, afaik.